### PR TITLE
DOC: Fix PEP-8 issues in indexing-, missing_data-, options- and release.rst

### PR DESCRIPTION
**Pull request by FHaase**

### CODE REVIEW

File: doc/source/indexing.rst [hunk 1]
```diff
@@ -1175,8 +1180,8 @@ If instead you don't want to or cannot name your index, you can use the name
 .. ipython:: python
    :suppress:
 
-   old_index = index
-   del index
+   old_index = index                                               # noqa: F821
+   del index                                                       # noqa: F821
```
Comment (datapythonista) [inline doc/source/indexing.rst]: if this has `:suppress:` and it's not visible, and it's broken to need the `F821`, I guess we can simply get rid of it. Makes sense?

File: doc/source/indexing.rst [hunk 2]
```diff
@@ -1307,7 +1312,7 @@ The ``in`` and ``not in`` operators
 
    try:
        old_d = d
-       del d
+       del d                                                       # noqa: F821
```
Comment (datapythonista) [inline doc/source/indexing.rst]: same as before, and I'm not sure it makes sense that `d` is not defined here to need the `noqa`, but it is in the previous line.

File: doc/source/indexing.rst [hunk 3]
```diff
@@ -1509,7 +1516,7 @@ default value.
 
 .. ipython:: python
 
-   s = pd.Series([1,2,3], index=['a','b','c'])
+   s = pd.Series([1, 2, 3], index=['a', 'b', 'c'])
    s.get('a')               # equivalent to s['a']
```
Comment (datapythonista) [inline doc/source/indexing.rst]: do you mind leaving just two spaces before the comment, as in most places

File: doc/source/indexing.rst [hunk 4]
```diff
@@ -1829,7 +1838,7 @@ that you've done this:
    def do_something(df):
        foo = df[['bar', 'baz']]  # Is foo a view? A copy? Nobody knows!
        # ... many lines here ...
-       foo['quux'] = value  # We don't know whether this will modify df or not!
+       foo['quux'] = value  # We don't know whether this will modify df or not!  # noqa: E501, F821
```
Comment (datapythonista) [inline doc/source/indexing.rst]: I'd move the comment to the previous line, and avoid `E501`

**Review by datapythonista — COMMENTED** [review]
Looks good, added few comments.

The fixes to the header will conflict with #24086, which is likely to be merged first.

**Comment by datapythonista** [comment]
@FHaase can you address the review comments and fix the conflicts please?

File: doc/source/options.rst [hunk 5]
```diff
@@ -454,14 +456,14 @@ For instance:
...
-   s/1.e6
+   s / 1.e3
+   s / 1.e6
 
 .. ipython:: python
    :suppress:
    :okwarning:
 
-   pd.reset_option('^display\.')
+   pd.reset_option(r'^display.')
```
Comment (datapythonista) [inline doc/source/options.rst]: I don't think we want to delete the backslash, I guess this this string is used in a regular expression

File: doc/source/indexing.rst [hunk 6]
```diff
@@ -1785,17 +1771,17 @@ interpreter executes this code:
 
 .. code-block:: python
 
-   dfmi.loc[:, ('one', 'second')] = value
+   dfmi.loc[:, ('one', 'second')] = value                          # noqa: F821
```
Comment (datapythonista) [inline doc/source/indexing.rst]: what do you think of adding `value = 'z'` before this, so we don't need to have all the `F821`? Can be in a `:suppress:` block, or directly here, whatever you think it's better.

File: doc/source/indexing.rst [hunk 7]
```diff
@@ -248,10 +249,10 @@ new column. In 0.21.0 and later, this will raise a ``UserWarning``:
...
 .. code-block:: ipython
 
-    In[1]: df = pd.DataFrame({'one': [1., 2., 3.]})
-    In[2]: df.two = [4, 5, 6]
+    In [1]: df = pd.DataFrame({'one': [1., 2., 3.]})
+    In [2]: df.two = [4, 5, 6]
     UserWarning: Pandas doesn't allow Series to be assigned into nonexistent columns - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute_access
-    In[3]: df
+    In [3]: df
     Out[3]:
```
Comment (datapythonista) [inline doc/source/indexing.rst]: does the `Out` need a space before the `[` too?

File: doc/source/indexing.rst [hunk 8]
```diff
@@ -1163,25 +1167,12 @@ with the name ``a``.
 If instead you don't want to or cannot name your index, you can use the name
 ``index`` in your query expression:
 
-.. ipython:: python
-   :suppress:
-
-   old_index = index
-   del index
```
Comment (datapythonista) [inline doc/source/indexing.rst]: do you mind rendering the page and making sure the result is not changed by removing these? can't really understand why these blocks are needed, but may be I'm missing something

**Review by datapythonista — COMMENTED** [review]
looks good, couple of comments

File: doc/source/indexing.rst (hunk 7 above)
Comment (FHaase) [reply doc/source/indexing.rst]: Nope, If the `:` align vertical it's all good. I think it's done for visual beauty.

File: doc/source/indexing.rst (hunk 8 above)
Comment (FHaase) [reply doc/source/indexing.rst]: I removed the `:suppress:` which resulted in 
```
In [211]: old_index = index
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-211-27ea1dd9dc9d> in <module>
----> 1 old_index = index

NameError: name 'index' is not defined

In [212]: del index
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-212-b747d883ab88> in <module>
----> 1 del index

NameError: name 'index' is not defined
```
Same for the other (If the try block is removed as well). I doubt it makes any difference.

I noticed a `df.index` in the block above so I tried to replace it with that, but `del` is not defined.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @FHaase

**Comment by jreback** [comment]
can you merge master

**Pull request merged**